src/workspace/mod.rs

```diff
@@ -1893,6 +1922,31 @@ impl WorkspaceDirectory {
   }
 }
 
+/// Reads compilerOptions from tsconfig.json file next to the package.json
+/// See https://github.com/denoland/deno/issues/28455#issuecomment-2734956368
+#[allow(clippy::disallowed_methods)]
+fn compiler_options_from_ts_config_next_to_pkg_json<TSys: FsRead>(
+  sys: &TSys,
+  pkg_json: &PackageJson,
+) -> Option<ParsedTsConfigOptions> {
+  let mut ts_config_path = pkg_json.path.clone();
+  ts_config_path.set_file_name("tsconfig.json");
+  if let Ok(text) = sys.fs_read_to_string(&ts_config_path) {
+    if let Ok(url) = url_from_file_path(&ts_config_path) {
+      if let Ok(config) = ConfigFile::new(&text, url) {
+        if let Ok(options) = config.to_compiler_options() {
+          return Some(options);
+        }
+      }
+    }
+  }
+  log::warn!(
+    "Failed to read tsconfig.json from {}",
+    ts_config_path.display()
+  );
```

Comment:
Maybe we should include the error message at the end of this message so they can understand why it failed?